src/components/scenes/Staking/EarnScene.tsx

```diff
@@ -0,0 +1,168 @@
+import { cacheStyles, Theme, useTheme } from '../../services/ThemeContext'
+
+interface Props extends EdgeSceneProps<'earnScene'> {}
+
+export interface EarnSceneParams {}
+
+interface StakePolicyPosition {
+  stakePolicy: StakePolicy
+  stakePosition: StakePosition
+}
+
+interface StakePolicyMap {
+  [walletId: string]: { stakePolicyPositions: StakePolicyPosition[]; stakePlugins: StakePlugin[] }
+}
+
+export const EarnScene = (props: Props) => {
+  const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
+  const wallets = Object.values(currencyWallets)
+
+  const [stakePolicyMap, setStakePolicyMap] = React.useState<StakePolicyMap>()
+
+  const [displayWallets, setDisplayWallets] = React.useState<EdgeCurrencyWallet[]>(wallets)
+  const [positionWallets, setPositionWallets] = React.useState<EdgeCurrencyWallet[]>([])
+
+  const [isPortfolioSelected, setIsPortfolioSelected] = React.useState(false)
+  const [isLoading, setIsLoading] = React.useState(true)
+
+  const handleSelectEarn = useHandler(() => setIsPortfolioSelected(false))
+  const handleSelectPortfolio = useHandler(() => setIsPortfolioSelected(true))
+
+  useAsyncEffect(
+    async () => {
+      if (stakePolicyMap != null) return
+
+      const positionWallets = []
+      const policyMap: StakePolicyMap = {}
+
+      for (const wallet of wallets) {
+        // Get all available stake policies
+        const { pluginId } = wallet.currencyInfo
+        if (SPECIAL_CURRENCY_INFO[pluginId]?.isStakingSupported === true && ENV.ENABLE_STAKING) {
+          // For each wallet
+          const stakePolicyPositions: StakePolicyPosition[] = []
+
+          try {
+            const stakePlugins = await getStakePlugins(pluginId)
+            for (const stakePlugin of stakePlugins) {
+              const stakePolicies = stakePlugin.getPolicies({ wallet, currencyCode: wallet.currencyInfo.currencyCode })
+
+              // Check if there's open positions
+              for (const stakePolicy of stakePolicies) {
+                const stakePosition = await stakePlugin.fetchStakePosition({ stakePolicyId: stakePolicy.stakePolicyId, wallet, account })
+                stakePolicyPositions.push({ stakePolicy, stakePosition })
+
+                const allocations = getPositionAllocations(stakePosition)
+                const { staked, earned, unstaked } = allocations
+                if ([...staked, ...earned, ...unstaked].some(positionAllocation => !zeroString(positionAllocation.nativeAmount))) {
+                  positionWallets.push(wallet)
+                }
+              }
+            }
+
+            policyMap[wallet.id] = { stakePolicyPositions, stakePlugins }
+            setStakePolicyMap({ ...policyMap })
+            setPositionWallets(positionWallets)
+          } catch (e) {
+            console.error(String(e))
```

Comment:
First of all, you don't need `String()` around `e`. Secondly, shouldn't this be `showDevError`?